african-market-place/src/App.js

```diff
@@ -1,25 +1,46 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import { Route, Switch } from 'react-router-dom';
+
+import { LandingPage } from './components/pages/Landing';
+import { LoginPage } from './components/pages/Login';
+import { NotFoundPage } from './components/pages/NotFound';
+import { SignUpPage } from './components/pages/SignUp';
+
+export default function App() {
+  // The reason to declare App this way is so that we can use any helper functions we'd need for business logic, in our case auth.
+  // React Router has a nifty useHistory hook we can use at this level to ensure we have security around our routes.
+  // const history = useHistory();
+
+  const initialForm = {
+    username: '',
+    email: '',
+    password: '',
+    confirm: '',
+  };
+
+  const [form, setForm] = useState(initialForm);
+
+  const onChange = e => {
+    //pending
+    const [name, value] = e.target;
+    setForm({
+      ...form,
+      [name]: value,
+    });
+  };
```

Comment:
For this project, we want to keep app js as clean as possible. As unit 3 people are linking up pages and private routes App.js get a ton of traffic so having it free from code makes it easier to track stuff please put the functions in the components that use them,

Reply:
https://github.com/WaterMyPlants-TeamPlants/Front-End/blob/main/src/App.js This is our app.js from our last bw